eli5/permutation_importance.py

```diff
@@ -0,0 +1,61 @@
+"""
+A module for computing feature importances by measuring how score decreases
+when a feature is not available.
+
+A similar method is described in Breiman, "Random Forests", Machine Learning,
+45(1), 5-32, 2001 (available online at
+https://www.stat.berkeley.edu/%7Ebreiman/randomforest2001.pdf), with an
+application to random forests. It is known in literature as
+"Mean Decrease Accuracy (MDA)" or "permutation importance".
+"""
+from __future__ import absolute_import
+
+import numpy as np  # type: ignore
+from sklearn.utils import check_random_state  # type: ignore
+
+
+def iter_shuffled(X, columns_to_shuffle=None, pre_shuffle=False,
```

Comment:
I would rename `pre_shuffle` to `shared_permutation` or `shared_order`. `pre_shuffle` describes the implementation; `shared_permutation` aims to describe the semantics.